src/main/java/org/skife/jdbi/v2/unstable/ObjectIterator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.skife.jdbi.v2.unstable;
+
+import java.lang.reflect.Array;
+import java.util.Iterator;
+
+/**
+ * Implements Iterator for unidentified arrays that have been cast to Object. Note that its elements will be returned as Object, primitives included (will be autoboxed).
+ */
+class ObjectIterator implements Iterator<Object>
```

Comment:
This class actually iterates over arrays, not any arbitrary `Object` (even though that's not proven at compile time) so IMO the name is a little confusing.  Maybe `ReflectionArrayIterator` or something like that?